stats/dummy/collector_test.go

```diff
@@ -52,7 +52,9 @@ func TestCollectorCollect(t *testing.T) {
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 		go func() { c.Run(ctx) }()
-		time.Sleep(1 * time.Millisecond)
```

Comment:
Hmm, it's not good that there was a sleep here, I think that I will probably have to refactor the `Dummy` collector a bit... The whole `running` variable should probably go, it shouldn't be needed since we have the `context`...

Reply:
I agree about the `Dummy` collector, I have mentioned the issue here: https://github.com/loadimpact/k6/issues/469#issuecomment-375210712 

Changing it would avoid all the "subpar" changes I made to the tests in order to get them to work.

Reply:
Ah, I missed this comment before posting again in the general PR thread. OK, I'm convinced we should refactor the `Dummy` collector. Not sure if my suggestion with using a channel instead of a `Sample` slice is the best way to go about it, but I don't have any other ideas. 

Maybe we can just remove the `running` check with `panic()` for this pull request and create a new issue and PR for the actual `Dummy` collector refactoring... @robingustafsson?

Reply:
One idea is to change the `Dummy` collector implementation by adding a buffered channel to signal the running start and use that instead of the polling solution in the tests. That should get rid of the nastiness in there and we can refactor the rest in a separate PR.

Reply:
I'm not sure we actually need to signal anything, the dummy collector is used just for tests, so we basically just have to have an easy way to read the collected samples. I think that we could just replace the current `[]stats.Sample` with a `Samples` channel (buffered or not, both could be useful and the buffer size will be supplied when the dummy collector is created anyway) for stuffing the results in the collector and reading them outside with `range`. When the context is closed, close the `Samples` channel an that's that.

But that should probably be a part of another PR for #469 or for a new issue. Do you think that simply removing [this check](https://github.com/loadimpact/k6/blob/6619892111baa18987596b57b62fc1365c058f95/stats/dummy/collector.go#L58-L60) temporarily (before that other PR is created) will be enough so we don't need all of that nasty polling and sleep?

Reply:
I guess so. And I would have to change the [stats/dummy/collector_test.go](https://github.com/loadimpact/k6/blob/master/stats/dummy/collector_test.go) test file. That would fix the issue on #469 as well.